### DISCUSSION TITLE
Waiting time on large projects

### DISCUSSION THREAD

**Discussion by Horiatu** [community/community#6966] [Copilot Conversations] [answered]
I have a very large project (I have just large projects on going) In JS/TS.
The copilot icon on VSC has a blue clock over it and a sliding bar defilates under "Copilort: Application Explorer" title for more than an hour.

I did not have any chance to see it working.

It would be nice to tell us what is doing in that huge empty panel, even better to give an estimate of the remaining time.

(Is it analysing my whole project? Why not just the current file?! Why does it not give a choice on what I want it to do?! I'm confused!!!)

[linecounter.txt](https://github.com/github/feedback/files/7438452/linecounter.txt)
.
![image](https://user-images.githubusercontent.com/7327557/139364215-4ba66735-af94-4fc6-9ba0-34bfe05c6871.png)

The refresh icon does not work (do nothing "command 'copilot.explorer.refresh' not found" - why is that even there?!)

**Comment by davecheney** [comment]
that may be intercepting your HTTPS connections?

**Reply by Horiatu** [reply]
?

**Reply by davecheney** [reply]
I’m sorry, my answer was truncated. What I meant to ask is there a firewall, vpn, or other device that may be interceding on your network traffic.

**Reply by Horiatu** [reply]
Fiddler doesn't show anything wrong.

**Comment by Krzysztof-Cieslak** [comment, marked as answer]
`Copilot: Application Explorer` is not provided by the GitHub Copilot extension. As far as I understand, this is coming from the "Copilot for VS Code" extension (https://marketplace.visualstudio.com/items?itemName=Metatype.copilot-vscode) why is some JS/TS development tool.

**Reply by Horiatu** [reply]
Sorry: makes no sense!
The only instructions available to interact with Copilot are through VS Code. As per se, the only interface provided to interact with Copilot is VS Code. 
As early testers we've been asked to provide feedback on any issues, and my first issue is that it is not working (under the provided interface) and the interface is not giving feedback why and how to rezolve.

**Reply by Horiatu** [reply]
From the emailled invitation to join it goes directly under VS Code and open the extension from Market Place. It Says:
"GitHub Copilot
GitHub Copilot is an AI pair programmer which suggests line completions and entire function bodies as you type. GitHub Copilot is powered by the OpenAI Codex AI system, trained on public Internet text and billions of lines of code.

To learn more about GitHub Copilot, visit copilot.github.com.

Technical Preview
Access to GitHub Copilot is limited to a small group of testers during the technical preview of GitHub Copilot. If you don’t have access to the technical preview, you will see an error when you try to use this extension."

After successful instalation (no error shown, plus message for accepting me in the small group of testers,) the extension appears in VS Code interface as predicted, but does not work and does not explain why.

**Reply by Krzysztof-Cieslak** [reply]
I'm not sure how I can be clearer here. The `Copilot: Application Explorer` view that is shown on your screenshot, and the` copilot.explorer.refresh` command that you mentioned in the original message are not provided by the GitHub Copilot extension.

As mentioned in my message, you probably have installed the wrong extension.
* Our extension is called `GitHub Copilot`, is published by `GitHub`, has ID `github.copilot`, and is here: https://marketplace.visualstudio.com/items?itemName=GitHub.copilot

* Extension that you have probably installed (judging by the screenshot and description) is called `Copilot for VSCode`, is published by `Metatype`, has ID `Metatype.copilot-vscode`, and is here: https://marketplace.visualstudio.com/items?itemName=Metatype.copilot-vscode

I do recommend checking our documentation at https://github.com/github/copilot-docs
